src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -262,26 +274,47 @@ public ResponseEntity<byte[]> getCourseIcon(@PathVariable Long courseId, @PathVa
+     * @param filename The filename of the file
      * @return The generated access token
      */
-    @GetMapping("files/attachments/access-token/{filename:.+}")
+    @GetMapping("files/file-upload-exercises/{exerciseId}/submissions/{submissionId}/{filename:.+}/access-token")
     @PreAuthorize("hasRole('USER')")
-    public ResponseEntity<String> getTemporaryFileAccessToken(@PathVariable String filename) {
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        // TODO: we should check that the filename actually exists in the context in Artemis, ideally we also verify that the user has access to this file already here
-        if (filename == null) {
+    public ResponseEntity<String> getTemporaryFileUploadExerciseSubmissionAccessToken(@PathVariable Long exerciseId, @PathVariable Long submissionId,
+            @PathVariable String filename) {
+        log.debug("REST request to get an access_token for a file upload exercise submission with exerciseId: {} and submissionId {}", exerciseId, submissionId);
+
+        Optional<FileUploadSubmission> optionalSubmission = fileUploadSubmissionRepository.findById(submissionId);
+        Optional<FileUploadExercise> optionalFileUploadExercise = fileUploadExerciseRepository.findById(exerciseId);
+        if (optionalSubmission.isEmpty() || optionalFileUploadExercise.isEmpty()) {
             return ResponseEntity.badRequest().build();
         }
-        try {
-            String temporaryAccessToken = tokenProvider.createFileTokenWithCustomDuration(authentication, 30, filename);
-            return ResponseEntity.ok(temporaryAccessToken);
+
+        FileUploadSubmission submission = optionalSubmission.get();
+        // user that submitted the exercise
+        Optional<User> userOfTheSubmission = ((StudentParticipation) submission.getParticipation()).getStudent();
+        if (userOfTheSubmission.isEmpty()) {
+            return ResponseEntity.badRequest().build();
         }
-        catch (IllegalAccessException e) {
+
+        User requestingUser = userRepository.getUser();
+        // auth check - either the user that submitted the exercise or the requesting user is at least a tutor for the exercise
+        if (!(Objects.equals(requestingUser, userOfTheSubmission.get())) && !authCheckService.isAtLeastTeachingAssistantForExercise(optionalFileUploadExercise)) {
             throw new AccessForbiddenException();
         }
+
+        Claims claims = Jwts.claims();
+        claims.put(TokenProvider.EXERCISE_ID_KEY, exerciseId);
+        claims.put(TokenProvider.SUBMISSION_ID_KEY, submissionId);
+        claims.put(TokenProvider.FILENAME_KEY, TokenProvider.DOWNLOAD_FILE + filename);
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        String temporaryAccessToken = tokenProvider.createFileTokenWithCustomDuration(authentication, 30, claims);
```

Comment:
The duration of 30 is used multiple times, consider moving it to a constant.

Reply:
Added a constant